### DISCUSSION TITLE
R buttom

### DISCUSSION THREAD

**Discussion by jjff2** [jilleb/MQB-FPA#59] [Modified profiles]
hi do anyone knows how to make r buttom work? I've heard that it should edit GW dataset. please advise. Thanks a lot.

**Comment by jjff2** [comment]
> hi do anyone knows how to make r buttom work? I've heard that it should edit GW dataset. please advise. Thanks a lot.

sorry i just find the solution in template

**Comment by Cuzoe** [comment]
Care to share for others in the future?

**Reply by jjff2** [reply]
Hi as I written. There is description in the jilleb’s template at the how mode buttom work.

**Reply by rkan22** [reply]
can you say more please

**Reply by jjff2** [reply]
> can you say more please

as i written, it's in the template. 
if you are hard finding this it's at the part of "ubyte enable_fpa_button_part".

**Reply by rkan22** [reply]
ok thanks you for reply. i tested not working

**Reply by jjff2** [reply]
Oh sorry. I didn’t buy the R button at last. I choose to use Audi R8 button.

**Comment by olli991** [comment]
You need to edit a bunch of variables also many of the unknown ones so far. It is possible to activate by copying all of that from arteon for example but it's not fully know what exactly is enabling that behavior. The time will tell hopefully

**Reply by lito69** [reply]
> Necesita editar un montón de variables y también muchas de las desconocidas hasta ahora. Es posible activar copiando todo eso de arteon, por ejemplo, pero no se sabe exactamente qué está habilitando ese comportamiento. el tiempo lo dira ojala

If arteon dataset is put, does it work?  Could you upload arteon dataset that you investigate?

**Reply by olli991** [reply]
If you falsh the full Arteon R, Tiguan R dataset the R button is workling, also with longpush functionality.

Today i was able to implement the values into a Cupra Leon dataset to enable the retrofitted KLR  steering wheel with profile button.. also Longpush is now working on there.
We now know the needed values but so far we don't know the connections between those... so not completly logical so far.

**Reply by olli991** [reply]
Attached the OEM datasets of those cars with buttons:
[TC00-Cupra-Ateca.zip](https://github.com/jilleb/MQB-FPA/files/9541503/TC00-Cupra-Ateca.zip)
[PH00_3Q0907530AL_V03935343NG_5463__Arteon_R_mit DCC.zip](https://github.com/jilleb/MQB-FPA/files/9541504/PH00_3Q0907530AL_V03935343NG_5463__Arteon_R_mit.DCC.zip)
[PF00-3Q0907530AG-Tiguan-R.zip](https://github.com/jilleb/MQB-FPA/files/9541505/PF00-3Q0907530AG-Tiguan-R.zip)

**Reply by lito69** [reply]
Edit a golf dataset to test if the button mode on the steering wheel works, does anyone want to try it?
[19_3q0907530AC_golf GTI AC SAK normal 2 sport 3 Off 5 steering comfort recordando el modo dsg y motor y HDC botón mode leon 4.zip](https://github.com/jilleb/MQB-FPA/files/9945510/19_3q0907530AC_golf.GTI.AC.SAK.normal.2.sport.3.Off.5.steering.comfort.recordando.el.modo.dsg.y.motor.y.HDC.boton.mode.leon.4.zip)

**Reply by JimVWGTE** [reply]
Any new updates ?

**Reply by Cuzoe** [reply]
I have modified my dataset with info from here... https://www.drive2.ru/l/717744973889412385/

I have not installed buttons with R button yet, so testing is still required.

**Reply by Mjukes** [reply]
> I have modified my dataset with info from here... https://www.drive2.ru/l/717744973889412385/
> 
> I have not installed buttons with R button yet, so testing is still required.

Hi @Cuzoe.   I wonder if you have any experience to share.  I'm flashing a Tiguan 4motion,  retrofitted with R button,  with the Tiguan R dataset.  Using Jilles editor I've mapped all the profiles.  However when I edit the button lists,  things go screwy.  If I leave the stock R lists in place,  the R button works,  but only scrolls through the limited profiles.  For Long press, as the car doesn't have  Race (6) I swap it for Sport in list 3 and  I've added 03 to F20,  but Long press doesn't work.

I just wondered if you (or anyone) understood the relationships between the button lists and the buttons actually working?

Many thanks

Mike

**Reply by djmartzian** [reply]
Hello @Mjukes,

Based on the below article offset _F20h_ seems to be wrong. What button type has been added to your car: **twister** or **_push_**? 
For the first, for a long click behavior - check **F2Ah** offset, for the latter check **_F2Bh_**.

Source: https://www.drive2.ru/l/722508608016755839/

Good luck

**Reply by Mjukes** [reply]
So we already have twist and have just added the R button but putting 03 (sport) in F2A doesn't work. 06 was there in the Stock R dataset. 

Also the stock R dataset was missing eco and comfort. I've mapped these but if I add them to the R button list, the button stops working. 

It's really weird.

**Reply by Mjukes** [reply]
Hi all. Hope @jilleb sees this, happy to send some beer tokens as I'm going insane. I'm adding R button to a Tiguan 4motion.

Firstly, long push NEVER works, even after mapping the profile to F20a. 
Secondly and most confusingly, is individual mode has Drive, Steering, ACC etc. 

As soon as I add the R button, Steering vanishes. 

I have even added the R button to the Tiguan 2019 dataset and the same thing happens. 

Is there anyone who could look at two datasets I have made which by all accounts should work, but don't

I either have

Steering shows. No R button

Or R button short press works, long press doesn't. Steering missing

I've spent days on this, checked and double checked everything and am all out of ideas and unfortunately I'm not smart enough to use Jilles 010 template (I've loaded it but it doesn't make sense to me). 

Can anyone help?  

Thanks

Mike

**Reply by djmartzian** [reply]
Hello @Mjukes,

Have you also tested by manipulating **F2Bh** offset?  Agree, that it doesn't make sense (with regards to above article), but one test more will not break anything.

Can you please share your modded datasets? I'll be happy to take a look.

Cheers,
D

**Reply by Mjukes** [reply]
Thanks D,  
So, somewhere between Jilles Tool and Hexed.it,  something has been changing bits without my knowledge.  I made a fresh dataset which resolved the Steering display issue,  but long press does not work.  

Someone suggested it could be a  Gateway SW issue 

System designation	GW MQB High
Software version	5463
Hardware version	524
VW/Audi part number	3Q0907530AL
Hardware part number	3Q0907530E


I've tried reading the template but it dumbfounds me.  any second opinion would be greatly appreciated.

Dataset attached
[PRIIT FPA_DATASET_ODIS_J600_260112100109.xml](https://github.com/user-attachments/files/24591063/PRIIT.FPA_DATASET_ODIS_J600_260112100109.xml)
 

Thank you so much.

Mike

**Reply by djmartzian** [reply]
Hello @Mjukes ,

I've tried to create a new dataset file. Basically from my knowledge so far, lines at offset  **07F0h** and **0800h** must be the same as Golf R datasets.  

[FPA_DATASET_ODIS_J600_260113210406_djmartzian.xml](https://github.com/user-attachments/files/24600197/FPA_DATASET_ODIS_J600_260113210406_djmartzian.xml)

Share your feedback after you tested it.

Good luck,
D

**Comment by JimVWGTE** [comment]
Thank you

Is there someone who can help me with this ?



Verzonden vanuit Outlook voor iOS<https://aka.ms/o0ukef>
________________________________
Van: Cuzoe ***@***.***>
Verzonden: Tuesday, December 30, 2025 9:28:56 AM
Aan: jilleb/MQB-FPA ***@***.***>
CC: JimVWGTE ***@***.***>; Comment ***@***.***>
Onderwerp: Re: [jilleb/MQB-FPA] R buttom (Discussion #59)


I have modified my dataset with info from here... https://www.drive2.ru/l/717744973889412385/

I have not installed buttons with R button yet, so testing is still required.

**Comment by djmartzian** [comment]
Hello everyone,

Based on the information from the article: https://www.drive2.ru/l/722508608016755839/ 
I have created a data-set for **MQB-Evo platform**( esp. _Skoda Octavia Mk4_): 
[DA_0019_7208_38K_MZ0N_010FPA000003_MQB37WKONV_voice_button_1push_sport.xml](https://github.com/user-attachments/files/24572854/DA_0019_7208_38K_MZ0N_010FPA000003_MQB37WKONV_voice_button_1push_sport.xml)


I haven't tested yet, so I'm sharing it for now. 

Cheers
